dnf/cli/commands/search.py

```diff
@@ -109,8 +111,24 @@ def _print_section_header(exact_match, attrs, keys):
         if not self.base.conf.showdupesfromrepos:
             limit = self.base.sack.query().filterm(pkg=counter.keys()).latest()
 
+        counter_sorted = counter.sorted(reverse=True, limit_to=limit)
+
+        if self.opts.alpha:
+            # first sort alphabetically
+            # then divide by groups that match (description, url, name)
+            # then sort groups by number of match
```

Comment:
It would probably be a bit more clear to put each of the three comment lines above the respective code it belongs to.

also:
> then sort groups by number of **matches**